README.md
# high_availability_linux


releasenotes.txt
#
#
# Author: $Author$
# Revision: $Revision$
# Date: $Date$
#

Fixes in rev. 29
- added removal line for load.sh to remove previous file
- resolved issue where IPTables redirect would persist after fail over
- updated logic to remove incron.secondary file on passive member if failover occurs
- added blocked port 25 for outgoing e-mails on the secondary to prevent e-mail notifications
- corrected syntax issue for primary check in skybox_watchdog.sh
- added DISABLE_FIREWALL flag to allow all traffic or a mix of all traffic, except gui. 
- fixed design issue with crontab - now unset will keep customer cronjobs
- fixed issue with value not checked on secondary server for incron.primary
- added lines to remove incron.primary in unset.sh
- updated $NOW variable in the filename creation

Feb 20, 2019
Fixes in rev. 28
- add directory symmetry for fw_config and ticket_attachments folder
- added task schedule disabling to conf_files_sync.sh script
- updated load.sh to include sqlx restore of user/model tables
- changed default timeout values in skybox_ha.conf
- changed SOAP ping to REST ping
- added capability to install in other directories, besides /home/skyboxview
- corrected MAIL_RECiVER spelling to MAIL_RECEIVER in skybox_watchdog.sh
- corrected issue found in skybox_watchdog.sh
- removed unnecessary files
- added logic to test if the user is skyboxview for setprimary.sh and setsecondary.sh to prevent scripts from running as other users, like root
- removed disabling task scheduling from conf_files_sync.sh
- moved Skybox logs and home location to skybox_ha.conf for easier configuration
- add new proxy settings for curl command
- fixed missing incron tab
- updated additional logic to setsecondary.sh - FD
- corrected unset.sh script - append flag was missing
- setprimary.sh - corrected MAIL_RECEIVER variable
- added test email notification when mail parameters are set correctly

====================================================================================================================================

May 22, 2018
Fixes in rev. 27
- fixed load.sh script, updated with better logic and logging
- added new field to skybox_ha.conf - DISABLE_SECONDARY_GUI
- changed default OTHER_SERVER ip address to a default string
- added logic to setsecondary.sh to load new iptables.secondary-no-gui tables in firewall directory
- added logic to setsecondary.sh to load new block_secondary_gui.xml tables in firewall director for firewalld service
- put .gitignore list on svn ignore list, as it's not needed
- added .idea directory to svn ignore list, as it's not needed for HA

====================================================================================================================================

May 16, 2018

Fixes in rev. 26
- conf_files_sync.sh script updated with copying sb_server.properties and sb_common.properties to configuration folder


====================================================================================================================================


March 21, 2018

Fix in rev. 25:
- Erroneous character found in skybox_watchdog.sh, removed character
- Removed soap_ping.pl and soap_ping.py
- Updated skybox_watchdog.sh with the function to perform the soap ping as part of its script
- Add req.xml (this is part of the API ping request)
- Updated setprimary.sh logic to allow the secondary to become active, even if there is an incron.primary file on the Primary server
- Correct redirect statement for setsecondary.sh script
- Path for mysqldump backupTicket.sh updated to reflect proper path depending on installed version



====================================================================================================================================

Dec 27, 2017
	
Fixes in rev. 24:
- Updated the MAIL_RECEIVER in skybox_ha.conf. The variable was misspelled causing other scripts to fail

Fixes of rev. 23:
- Updated the e-mail check in setprimary.sh. The false notification was caused by script was running. The proper logic has been put in place to check if the mail parameters have  been set.
- Updated skybox_watch.sh with the proper e-mail notification function. Only in the event of a failover will the e-mail be sent. I have tested and verified that this is working.
- Updated setprimary.sh, setsecondary.sh, and skybox_watchdog.sh with better logging logic. The code is cleaner and nicer to read


load.sh
#!/bin/bash
#
#
# Author: $Author$
# Revision: $Revision$
# Date: $Date$
#

# global variables - start

source `dirname $0`/skybox_ha.conf

FILENAME=${1##*/}

# global variables - stop
#add logic for errors - file not match or blank

# main code - start
cd $SKYBOX_HOME/server/bin

# check if last 5 characters equals sqlx, if so, use sqlxrestore
if [[ ${1: -5} == ".sqlx" ]] 
then
        echo sqlxrestore.sh $1 -model
        #echo "TIME=$(date)" > $MODELSTATUS
        date +"%Y-%m-%d %T - FILENAME=$1" | tee -a $MODELSTATUS

        if ./sqlxrestore.sh -user_tables -model_tables $1 2>&1 >> $SKYBOXLOG; then
            date +"%Y-%m-%d %T - Restoration of sql file: $FILENAME - SUCCESS" | tee -a $MODELSTATUS
		    date +"%Y-%m-%d %T - Restoration of sql file: $FILENAME - SUCCESS" | tee -a $SKYBOXLOG
        else
            date +"%Y-%m-%d %T - Restoration of sql file: $FILENAME - FAILED" | tee -a $MODELSTATUS
		    date +"%Y-%m-%d %T - Restoration of sql file: $FILENAME - FAILED" | tee -a $SKYBOXLOG
        fi

# check if the file name is ticket_backup, if so, load via mysql.sh
elif [[ ${1%*.*} == "/opt/skyboxview/data/xml_models/ticket_backup.sql" ]]; then
            echo "mysql.sh $1"
            #echo "TIME=$(date)" > $LOADSTATUS
             date +"%Y-%m-%d %T - FILENAME=$1" >> $LOADSTATUS
             ./mysql.sh --database=skyboxview_live --password=manager < $1 2>&1 | tee -a $SKYBOXLOG

            if [ $? == 0 ]; then
                ./invalidate.sh
                 date +"%Y-%m-%d %T -  Restoration of backup_sql file: $FILENAME - SUCCESS" | tee -a $LOADSTATUS
			     date +"%Y-%m-%d %T -  Restoration of backup_sql file: $FILENAME - SUCCESS" | tee -a $SKYBOXLOG
				 
				 if rm -f $1; then
				     date +"%Y-%m-%d %T -  Removing of ticket backup file: $FILENAME - SUCCESS" | tee -a $LOADSTATUS
			         date +"%Y-%m-%d %T -  Restoration of backup_sql file: $FILENAME - SUCCESS" | tee -a $SKYBOXLOG
				else
					 date +"%Y-%m-%d %T -  Restoration of backup_sql file: $FILENAME - FAILURE" | tee -a $LOADSTATUS
			         date +"%Y-%m-%d %T -  Restoration of backup_sql file: $FILENAME - FAILURE" | tee -a $SKYBOXLOG
				fi
				
            else
                 date +"%Y-%m-%d %T - Restoration of backup_sql file: $FILENAME - FAILED" |  tee -a $LOADSTATUS
			     date +"%Y-%m-%d %T - Restoration of backup_sql file: $FILENAME - FAILED" | tee -a $SKYBOXLOG
            fi
# check if the last 5 characters of filename is xmlx, then load model using xmlx load.sh script
else
        if [[ ${1: -5} == ".xmlx" ]] ; then
		     
             echo "load.sh $1 -model"
             date +"%Y-%m-%d %T - FILENAME=$1" | tee -a $MODELSTATUS

                if ./load.sh $1 -model -coreusers -core | tee -a $SKYBOXLOG ; then
				    date +"%Y-%m-%d %T - Model Upload Status - $FILENAME - SUCCESS" | tee -a $MODELSTATUS
				    date +"%Y-%m-%d %T - Model Upload Status - $FILENAME - SUCCESS" | tee -a $SKYBOXLOG
                else
                    date +"%Y-%m-%d %T - Model Upload Status - $FILENAME - FAILED" | tee -a $MODELSTATUS
				    date +"%Y-%m-%d %T - Model Upload Status - $FILENAME - FAILED" | tee -a $SKYBOXLOG
                fi
        fi
fi

# main code - stop

unsetha.sh
#!/bin/bash
#
#
# Author: $Author$
# Revision: $Revision$
# Date: $Date$
#

# source the configuration file

# global values - start
# global variables - end


## declare an array variable
declare -a PRIMARYARRAY=("skybox_watchdog.sh" "backupTicket.sh" "replicate_fw_config.sh" "replicate_ticket_attachments.sh")
declare -a SECONDARYARRAY=("skybox_watchdog.sh" "conf_files_sync.sh")
## end of declaration

source `dirname $0`/skybox_ha.conf

# remove only Skybox crontab lines
#crontab -u mobman -l | grep -v 'perl /home/mobman/test.pl'  | crontab -u mobman -

if [ -e "$SKYUSERHOME/incron.primary" ]; then
    for VAL in "${PRIMARYARRAY[@]}"; do
       crontab -l | grep -v $VAL | crontab -
    done
else
    for VAL in "${SECONDARYARRAY[@]}"; do
      crontab -l | grep -v $VAL | crontab -
    done
fi

if [ -e "$SKYUSERHOME/incron.primary" ]; then
    PRIMARYTESTRESULTS=`crontab -l | grep -q 'skybox_watchdog.sh|backupTicket.sh|replicate_fw_config.sh|replicate_ticket_attachments.sh' | wc -l`
    if [ "$PRIMARYTESTRESULTS" -eq '0' ]; then
        echo `date +"%Y/%m/%d %H:%M:%S - Crontab cleared successfully"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Crontab failed to be cleared - try again"` | tee -a $SKYBOXLOG
    fi
else

    SECONDARYTESTRESULTS=`crontab -l | grep -q 'skybox_watchdog.sh|conf_files_sync.sh' | wc -l`
    if [ "$SECONDARYTESTRESULTS" -eq '0' ]; then
        echo `date +"%Y/%m/%d %H:%M:%S - Crontab cleared successfully"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Crontab failed to be cleared - try again"` | tee -a $SKYBOXLOG
    fi
fi


# delete all iptables
sudo iptables -t nat -F
echo "Removing current IPtables configuration" | tee -a $SKYBOXLOG

# create blank incron.primary and secondary files
if [ -f "$SKYUSERHOME/incron.primary" ]
then
    echo '' > $SKYUSERHOME/incron.primary
    echo `date +"%Y/%m/%d %H:%M:%S - Clearing incron.primary"` | tee -a $SKYBOXLOG
    incrontab $SKYUSERHOME/incron.primary
    if [ $? == 0 ]
    then
	    echo `date +"%Y/%m/%d %H:%M:%S - Incrontab on primary cleared successfully"` | tee -a $SKYBOXLOG
    else
   	    echo `date +"%Y/%m/%d %H:%M:%S - Incrontab on primary failed to be cleared"` | tee -a $SKYBOXLOG
    fi
elif [ -f "$SKYUSERHOME/incron.secondary" ]
then
    echo '' > $SKYUSERHOME/incron.secondary
    incrontab $SKYUSERHOME/incron.secondary
    if [ $? == 0 ]
    then
    	echo `date +"%Y/%m/%d %H:%M:%S - Incrontab on secondary cleared successfully"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Incrontab on secondary failed to be cleared"` | tee -a $SKYBOXLOG
    fi
else
    echo `date +"%Y/%m/%d %H:%M:%S - No incron files were found!"` | tee -a $SKYBOXLOG
fi

if (( $(service iptables status | grep Active | awk '{ print $2 }' | wc -l) > 0 )); then
    echo "IPtables service is running, prepraing to load configuration"
    sudo /sbin/iptables-restore < $SKYUSERHOME/firewall/iptables-secondary
    if [ $? == 0 ]
    then
        echo `date +"%Y/%m/%d %H:%M:%S - Iptables configuration was uploaded..."` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Iptables configuration failed to load..."` | tee -a $SKYBOXLOG
        state=1
    fi
elif (( $(service firewalld status | grep Active | awk '{ print $2 }' | wc -l) > 0 )); then
    echo "Firewalld detected..."
    # remove direct.xml to open port 25

    if [ -f "/etc/firewalld/direct.xml" ]; then
        sudo rm /etc/firewalld/direct.xml
    fi


	# copy back original public configuration
    sudo cp $SKYUSERHOME/firewall/original_public.xml /etc/firewalld/zones/public.xml
    sudo firewall-cmd --reload
    if [ $? == 0 ]
    then
        echo `date +"%Y/%m/%d %H:%M:%S - Firewalld configuration was uploaded,firewall was successfully reloaded"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Firewalld configuration was uploaded, but firewall was not reloaded"` | tee -a $SKYBOXLOG
        state=1
    fi
else
    echo `date +"%Y/%m/%d %H:%M:%S - This is not a recognized version"` | tee -a $SKYBOXLOG
fi

# set task scheduler according to server role in skybox_ha.conf_files_sync
case $SERVER_ROLE in
    1)
    cd $SKYBOX_HOME/server/bin/
    if ./settaskschedulingactivation.sh enable >/dev/null; then
        echo `date +"%Y/%m/%d %H:%M:%S - Task Scheduling has been successfully enabled."` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Task Scheduling was not enabled. An error has occurred."` | tee -a $SKYBOXLOG
    fi

    ;;

    2)
    cd $SKYBOX_HOME/server/bin/
    if ./settaskschedulingactivation.sh disable >/dev/null; then
        echo `date +"%Y/%m/%d %H:%M:%S - Task Scheduling has been successfully disabled."` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y/%m/%d %H:%M:%S - Task Scheduling was not disabled. An error has occurred."` | tee -a $SKYBOXLOG
    fi
    ;;

        *)
                echo `date +"%Y/%m/%d %H:%M:%S - Server options are not set correctly. Configure the skybox_ha.conf with the correct SERVER_ROLE"` | tee -a $SKYBOXLOG
        ;;
esac

# removing incron.primary or incron.secondary
# removing incrontab files and lllltabs
rm -f $SKYUSERHOME/incron.primary
rm -f $SKYUSERHOME/incron.secondary
incrontab -r


backupTicket.sh
#!/bin/bash
#
#
# Author: $Author$
# Revision: $Revision$
# Date: $Date$
#

# source the configuration file
source `dirname $0`/skybox_ha.conf

# global variables - start

prifile="$SKYUSERHOME/incron.primary"
MYSQLDUMP=`find /opt/skyboxview/thirdparty -path "*bin/mysqldump"`

# global variables - stop

# script start

# Check if this server is either Primary or secondary (if not exist)
 if [ -e $prifile ]; then
     echo "Production Primary Skybox Server Ok"
 
 #I'm primary export ticket
NOW=$(date +"%Y%d%m%H%M%S")
$MYSQLDUMP --user root --password=manager -S /opt/skyboxview/data/db/mysql/mysql.sock skyboxview_live sbv_tickets sbv_ticket_events sbv_ticket_fields sbv_attachments sbv_phases > /opt/skyboxview/data/xml_models/ticket_backup.sql.$NOW
    echo $NOW
    echo `date +"%Y/%m/%d %H-%M-%S - Backup tickets finished successfully"` | tee -a $SKYBOXLOG
else
    echo "Not Primary Skybox Server"
    exit 255
fi


# script stop

conf_files_sync.sh
#!/bin/bash
#
#
# Author: $Author$
# Revision: $Revision$
# Date: $Date$
#


# global variables - start

# sourcing configuration file
source `dirname $0`/skybox_ha.conf

#global variables - stop

# functions - start

#funcDisableTaskScheduler(){
#cd $SKYBOX_HOME/server/bin/
												   
#if ./settaskschedulingactivation.sh disable >/dev/null; then
#        echo `date +"%Y-%m-%d %T - Task Scheduling has been disabled"` | tee -a $SKYBOX_HOME/server/log/debug/debug.log $SKYBOXLOG
#    else
#        echo `date +"%Y-%m-%d %T - ERROR: Task Scheduling has not been set to false. Please rerun the setsecondary script again."` |
 #       tee -a  $SKYBOX_HOME/server/log/debug/debug.log $SKYBOXLOG
#fi
#}


# functions - stop

# main code - start

# scp file from other server to /tmp folder; copy over files from temp to skybox conf directory
if `scp skyboxview@$OTHER_SERVER:$SKYBOX_HOME/server/conf/sb_server.properties /tmp/sb_server.properties`; then
    echo `date +"%Y/%m/%d %H:%M:%S - sb_server.properties copied to temp. Preparing to copy the file..."` | tee -a $SKYBOXLOG
	if `sed -i "s/task_scheduling_activation\=true/task_scheduling_activation\=false/g"  /tmp/sb_server.properties`; then
	    echo `date +"%Y/%m/%d %H:%M:%S -  keep_temp flag was successfully changed to false."` | tee -a $SKYBOXLOG
			if `cp /tmp/sb_server.properties $SKYBOX_HOME/server/conf/sb_server.properties`; then
	            echo `date +"%Y/%m/%d %H:%M:%S - sb_server.properties copied to server conf diretory."` | tee -a $SKYBOXLOG
		        rm /tmp/sb_server.properties
            else
	            echo `date +"%Y/%m/%d %H:%M:%S - sb_server.properties was not copied over"` | tee -a $SKYBOXLOG
	        fi
	else
	    `date +"%Y/%m/%d %H:%M:%S - keep_temp was not changed!."` | tee -a $SKYBOXLOG
	fi
	

else
   echo `date +"%Y/%m/%d %H:%M:%S - sb_server.properties was not copied over. Verify scp settings"` | tee -a $SKYBOXLOG
fi

# 
if `scp skyboxview@$OTHER_SERVER:$SKYBOX_HOME/server/conf/sb_common.properties /tmp/sb_common.properties`; then
    echo `date +"%Y/%m/%d %H:%M:%S - sb_common.properties copied. Preparing for next step..."` | tee -a $SKYBOXLOG
		if `cp /tmp/sb_common.properties $SKYBOX_HOME/server/conf/sb_common.properties`; then
	    echo `date +"%Y/%m/%d %H:%M:%S - sb_common.properties copied to server conf diretory."` | tee -a $SKYBOXLOG 
		rm /tmp/sb_common.properties
    else
	    echo `date +"%Y/%m/%d %H:%M:%S - sb_common.properties was not copied over"` | tee -a $SKYBOXLOG
	fi
else
   echo `date +"%Y/%m/%d %H:%M:%S - sb_common.properties was not copied over. Verify scp settings"` | tee -a $SKYBOXLOG
fi

# main code - stop


setsecondary.sh
#!/bin/bash
#
#
# Author: $Author: frank.bailey $
# Revision:$Revision: 28 $
# Date: $Date: 2019-02-21 09:59:54 -0500 (Thu, 21 Feb 2019) $
#
# Additional Logic provided by Filippo Dino
#

# global variables - start
source `dirname $0`/skybox_ha.conf
TESTFIREWALLD=`ps -ef | grep -v grep | grep firewalld | wc -l`
state=0

# global variables - end

# main code - start

# logic to verify if the user is skyboxview; else exits
if [ `whoami` == 'skyboxview' ]; then
    echo "The user is $(whoami). Proceeding to set as secondary..." | tee -a $SKYBOXLOG
else
    echo " The user is $(whoami) . Script must be ran as the skyboxview user. Exiting script..." | tee -a $SKYBOXLOG
    exit 1
fi

# verify if the ticket_attachments folder exists
if [ -d $SKYBOX_HOME/data/ticket_attachments ]; then
    echo "$SKYBOX_HOME/data/ticket_attachments exists"
else
    echo "$SKYBOX_HOME/data/ticket_attachments doesn't exist. Creating ticket_attachments folder..."
	if `mkdir $SKYBOX_HOME/data/ticket_attachments`; then
	    echo `date +"%Y-%m-%d %T - $SKYBOX_HOME/data/ticket_attachments was created successfully"` | tee -a $SKYBOXLOG
	else
	    echo `date +"%Y-%m-%d %T - $SKYBOX_HOME/data/ticket_attachments failed to be created. Please check folder permissions"` | tee -a $SKYBOXLOG
    fi
fi

# removing incrontab files and tabs
if [ -f "$SKYUSERHOME/incron.primary" ]
then
    echo "Removing incron.primary..."
    rm -f $SKYUSERHOME/incron.primary
    if [ $? == 0 ]
    then
        echo `date +"%Y-%m-%d %T - Incron.primary file was removed successfully"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Incron.primary file failed to be removed"` | tee -a $SKYBOXLOG
    fi
elif [ -f "$SKYUSERHOME/incron.secondary" ]
then
    echo "Removing incron.secondary file..."
    rm -f $SKYUSERHOME/incron.secondary
    if [ $? == 0 ]
    then
        echo `date +"%Y-%m-%d %T - Incron.secondary file was removed successfully"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Incron.secondary file failed to be removed"` | tee -a $SKYBOXLOG
    fi
else
    echo `date +"%Y-%m-%d %T - No incron files were found!"` | tee -a $SKYBOXLOG
fi

#removing incrontab
incrontab -r

# adding cron entries for secondary
echo $SKYBOX_HOME/data/xml_models IN_MOVED_TO $SKYUSERHOME/load.sh \$@/\$\# >> $SKYUSERHOME/incron.secondary
echo $SKYBOX_HOME/data/sqlx_models IN_MOVED_TO $SKYUSERHOME/load.sh \$@/\$\# >> $SKYUSERHOME/incron.secondary
incrontab $SKYUSERHOME/incron.secondary
incrontab -d
#sudo /sbin/iptables-restore < $SKYUSERHOME/iptables-secondary
#sudo service iptables restart
if [ "$1" == "auto" ]; then
        echo "Resetting skybox crontab and adding watchdog script"
        crontab -l > $SKYUSERHOME/cron
        sed -i '/skybox_watchdog.sh/d' $SKYUSERHOME/cron
        sed -i '/backupTicket.sh/d' $SKYUSERHOME/cron
        sed -i '/replicate_fw_config.sh/d' $SKYUSERHOME/cron
        if ! (grep -Fxq '*/$REST_PING_INTERVAL * * * * $SKYUSERHOME/skybox_watchdog.sh >/dev/null 2>&1' $SKYUSERHOME/cron)
        then
            echo "*/$REST_PING_INTERVAL * * * * $SKYUSERHOME/skybox_watchdog.sh >/dev/null 2>&1" >> $SKYUSERHOME/cron
        else
            date +"%Y-%m-%d %T - Watchdog Cron entry already exist" | tee -a $SKYBOXLOG
        fi
        if !(grep -Fxq '*/60 * * * * $SKYUSERHOME/conf_files_sync.sh >/dev/null 2>&1' $SKYUSERHOME/cron)
        then
            echo "*/60 * * * * $SKYUSERHOME/conf_files_sync.sh >/dev/null 2>&1" >> $SKYUSERHOME/cron
        else
            echo `date +"%Y-%m-%d %T - Conf_files_sync Cron entry already exist"` | tee -a $SKYBOXLOG
        fi
        crontab $SKYUSERHOME/cron
        echo `date +"%Y-%m-%d %T - Setsecondary run with auto"` | tee -a $SKYBOXLOG
else
        echo "Removing skybox_watchdog.sh from crontab, auto failover is not configured"
        crontab -l > $SKYUSERHOME/cron
        sed -i '/skybox_watchdog.sh/d' $SKYUSERHOME/cron
	sed -i '/backupTicket.sh/d' $SKYUSERHOME/cron
	sed -i '/replicate_fw_config.sh/d' $SKYUSERHOME/cron
        crontab $SKYUSERHOME/cron
        echo "To enable auto failover, rerun this script as follows"
        echo "$0 auto"
        echo `date +"%Y/%m/%d %T - Setsecondary run without auto"` | tee -a $SKYBOXLOG
        state=1
fi

# Set firewalls back to original installation state

# set firewalld to block secondary gui id for firewalld with standard ports are blocked
if [ $TESTFIREWALLD -gt '0' ] && [ $DISABLE_FIREWALLS == "false" ] && [ $DISABLE_SECONDARY_GUI == "true" ]; then
    echo "Detected Firewalld, loading configuration..."
    echo "Disable Secondary Gui is set to true..."
    echo "Allow all ports is set to false..."
    
	# add port 25 block on direct interface
	sudo cp $SKYUSERHOME/firewall/direct.xml /etc/firewalld/
	
	# add block secondary gui xml to correct zones/public
    sudo cp $SKYUSERHOME/firewall/block_secondary_gui.xml /etc/firewalld/zones/public.xml
    sudo firewall-cmd --reload
    if [ $? == 0 ]
    then
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, firewall with secondary gui disabled was successfully reloaded"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, but firewall was not reloaded"` | tee -a $SKYBOXLOG
        state=1
    fi

# set firewalld to block secondary gui id for firewalld, but allow ALL other ports
elif [ $TESTFIREWALLD -gt '0' ] && [ $DISABLE_FIREWALLS == "true" ] && [ $DISABLE_SECONDARY_GUI == "true" ]; then
    echo "Detected Firewalld, loading configuration..."
    echo "Disable Secondary Gui is set to true..."
    echo "Allow all ports set to true" 
	
	# add block secondary gui xml to correct zones/public
    sudo cp $SKYUSERHOME/firewall/block_secondary_gui_allow_all.xml /etc/firewalld/zones/public.xml
    sudo firewall-cmd --reload
    if [ $? == 0 ]
    then
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, firewall with secondary gui and disable firewalls enabled was successfully reloaded"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, but firewall was not reloaded"` | tee -a $SKYBOXLOG
        state=1
    fi

# set firewalld to original configuration
elif [ $TESTFIREWALLD -gt '0' ] && [ $DISABLE_FIREWALLS == "false" ] && [ $DISABLE_SECONDARY_GUI == "false" ]; then
    echo "Detected Firewalld, loading configuration..."
    echo "Disable Secondary Gui is set to true..."
    echo "Allow all ports set to false..."
    
	# add port 25 block on direct interface
	sudo cp $SKYUSERHOME/firewall/direct.xml /etc/firewalld/
	
	
	# add original configuration for secondary
    sudo cp $SKYUSERHOME/firewall/original_public.xml /etc/firewalld/zones/public.xml
    sudo firewall-cmd --reload
    if [ $? == 0 ]
    then
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, firewall with secondary gui disabled was successfully reloaded"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, but firewall was not reloaded"` | tee -a $SKYBOXLOG
        state=1
    fi

# set firewalld to allow traffic due to disable_firewalls flag enabled to true
elif [ $TESTFIREWALLD -gt '0' ] && [ $DISABLE_FIREWALLS == "true" ] && [ $DISABLE_SECONDARY_GUI == "false" ]; then
    echo "Detected Firewalld, loading configuration..."
    echo "Disable Secondary Gui is set to true..."
	# add port 25 block on direct interface
	sudo cp $SKYUSERHOME/firewall/direct.xml /etc/firewalld/
	
	
	# add original configuration for secondary
    sudo cp $SKYUSERHOME/firewall/original_public_disable.xml /etc/firewalld/zones/public.xml
    sudo firewall-cmd --reload
    if [ $? == 0 ]
    then
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, firewall with secondary gui disabled, but disable all firewalls was successfully reloaded"` | tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Firewalld configuration was uploaded, but firewall was not reloaded"` | tee -a $SKYBOXLOG
        state=1
    fi


# if no gui is set to true, but disable_firewalls is set to false, disable port 8443 and allow normal traffic for IPTABLES
elif [ $TESTFIREWALLD -eq '0' ] && [ $DISABLE_FIREWALLS == "false" ] && [ $DISABLE_SECONDARY_GUI == "true" ]; then
    echo "Detected IPTables, loading configuration..."
	sudo iptables -t nat -F
    if sudo /sbin/iptables-restore < $SKYUSERHOME/firewall/iptables-secondary-no-gui; then
        echo `date +"%Y-%m-%d %T - Iptables configuration with no secondary gui only, but  was uploaded..."`| tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Iptables configuration with no secondary gui failed to load..."` | tee -a $SKYBOXLOG
        state=1
    fi

# if disable_firewalls is set to true, but no gui is set to false, allow all traffic for IPTABLES
elif [ $TESTFIREWALLD -eq '0' ] && [ $DISABLE_FIREWALLS == "true" ] && [ $DISABLE_SECONDARY_GUI == "false" ]; then
    echo "Detected IPTables, loading configuration..."
	sudo iptables -t nat -F
    if sudo /sbin/iptables-restore < $SKYUSERHOME/firewall/iptables-secondary-disable; then
        echo `date +"%Y-%m-%d %T - Iptables Disable_firewalls configuration  was uploaded..."`| tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Iptables configuration with no secondary gui failed to load..."` | tee -a $SKYBOXLOG
        state=1
    fi


# set gui and disable_firewalls is set to true, disable port 8443 and allow all traffic for IPTABLES
elif [ $TESTFIREWALLD -eq '0' ] && [ $DISABLE_FIREWALLS == "true" ] && [ $DISABLE_SECONDARY_GUI == "true" ]; then
    echo "Detected IPTables, loading configuration..."
	sudo iptables -t nat -F
    if sudo /sbin/iptables-restore < $SKYUSERHOME/firewall/iptables-secondary-no-gui-allow-all; then
        echo `date +"%Y-%m-%d %T - Iptables configuration with no secondary gui and disable all firewalls enabled was uploaded..."`| tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Iptables configuration with no secondary gui failed to load..."` | tee -a $SKYBOXLOG
        state=1
    fi

# set gui and firewalls to false, normal traffic for IPTABLES
elif [ $TESTFIREWALLD -eq '0' ] && [ $DISABLE_FIREWALLS == "false" ] && [ $DISABLE_SECONDARY_GUI == "false" ]; then
    echo "Detected IPTables, loading configuration..."
	sudo iptables -t nat -F
    if sudo /sbin/iptables-restore < $SKYUSERHOME/firewall/iptables-secondary; then
        echo `date +"%Y-%m-%d %T - Iptables configuration to  allow normal traffic was uploaded..."`| tee -a $SKYBOXLOG
    else
        echo `date +"%Y-%m-%d %T - Iptables configuration with no secondary gui failed to load..."` | tee -a $SKYBOXLOG
        state=1
    fi

else
    echo "There appears to a problem with the configuration, check the skybox_ha.conf"
    state=1
fi

# disabling task scheduler in sb_server.properties
cd $SKYBOX_HOME/server/bin/
												   
if ./settaskschedulingactivation.sh disable >/dev/null; then
        echo `date +"%Y-%m-%d %T - Task Scheduling has been disabled"` | tee -a $SKYBOX_HOME/server/log/debug/debug.log $SKYBOXLOG
        state=0
    else
        echo `date +"%Y-%m-%d %T - ERROR: Task Scheduling has not been set to false. Please rerun the setsecondary script again."` |
        tee -a  $SKYBOX_HOME/server/log/debug/debug.log $SKYBOXLOG
        state=1
fi

# disabling _collectors_auto_update flag in sb_server.properties
cd $SKYBOX_HOME/server/bin/
if grep -rni "enable_collectors_auto_update=true" $SKYBOX_HOME/server/conf/sb_server.properties
    then
        echo `date +"%Y-%m-%d %T - enable_collectors_auto_update flag is set to true, changing flag to false"` |
        tee -a $SKYBOX_HOME/server/log/debug/debug.log
        sed -i 's/enable_collectors_auto_update=true/enable_collectors_auto_update=false/g' $SKYBOX_HOME/server/conf/sb_server.properties
        if [ $? == 0 ]
            then
                echo `date +"%Y-%m-%d %T - enable_collectors_auto_update flag has been successfully set to false..."` |
                tee -a $SKYBOX_HOME/server/log/debug/debug.log $SKYBOXLOG
                state=0
            else
                echo `date +"%Y-%m-%d %T - enable_collectors_auto_update flag has not been set to false, please rerun script"` |
                tee -a $SKYBOX_HOME/server/log/debug/debug.log $SKYBOXLOG
                state=1
        fi
    else
        echo `date +"%Y-%m-%d %T - enable_collectors_auto_update flag is set to false, no change needed..."` |
        tee -a $SKYBOX_HOME/server/log/debug/debug.log  $SKYBOXLOG
fi

#
# checks state variable to see if all of the scripts were ran correctly
#
if [ $state == 0 ]
then
   echo `date +"%Y-%m-%d %T - Host has been successfully set as secondary"` | tee -a $SKYBOXLOG
else
   echo `date +"%Y-%m-%d %T - Host failed to be set as secondary"` | tee -a $SKYBOXLOG
fi


# main code - stop
